### PR TITLE
Clean: UndoStateShape indexing meshptr to string

### PR DESCRIPTION
**Pull request by sts1skj**
I changed the indexing of UndoStateShape in UndoStateProject (member
variable usss) to use a string shape name instead of a mesh pointer,
which should make it a bit more robust.

### CODE REVIEW

**Comment by sts1skj** [comment]
After this, I was thinking of adding x-mirror bone functionality.  The main question I have is where to put the new user-interface element.  I was thinking of making a drop-down menu with the options "None", "Automatic", and all of the project bones.  Maybe it would be best to put the drop-down menu in the brush settings pane, maybe somewhere else.

**Comment by ousnius** [comment]
@sts1skj I would do it like this:
image

The choice would need to be filled with the bone names everytime a bone is added, removed or the bone tree is reloaded.

The checkbox and choice need to be set to hidden/visible where the bone scale slider is set to hidden/visible as well.

Edit: Actually the "None" choice isn't needed, that would be disabling the checkbox.

File: src/program/OutfitStudio.cpp [hunk 1]
```diff
@@ -8932,15 +8936,12 @@ void wxGLPanel::EndPivotPosition() {
 	std::vector<mesh*> refMeshes = activeStroke->GetRefMeshes();
 	if (refMeshes.size() > 0) {
 		mesh* pivotHitMesh = refMeshes[0];
```
Comment (ousnius) [inline src/program/OutfitStudio.cpp]: "pivotHitMesh" (line 8938) is now unused. Can the line be removed?

Comment (sts1skj) [reply src/program/OutfitStudio.cpp]: Yes.  Shall I do it?

**Pull request merged**